modules/text2vec-databricks/clients/databricks_test.go

```diff
@@ -0,0 +1,302 @@
+			"neither in request header: X-Databricks-Token "+
+			"nor in environment variable under DATABRICKS_TOKEN")
+	})
+
+	t.Run("when X-Databricks-Servingurl header is passed", func(t *testing.T) {
+		c := New("", 0, nullLogger())
+
+		config := ent.VectorizationConfig{
+			Type:       "text",
+			Model:      "ada",
+			BaseURL:    "http://default-url.com",
+			ServingURL: "http://serving-url-in-config.com",
+		}
+
+		ctxWithValue := context.WithValue(context.Background(),
+			"X-Databricks-Servingurl", []string{"http://serving-url-passed-in-header.com"})
+
+		servingURL, err := c.buildURL(ctxWithValue, config)
+		require.NoError(t, err)
+		assert.Equal(t, "http://serving-url-passed-in-header.com", servingURL)
+
+		servingURL, err = c.buildURL(context.TODO(), config)
+		require.NoError(t, err)
+		assert.Equal(t, "http://serving-url-in-config.com", servingURL)
+	})
+
+	t.Run("pass rate limit headers requests", func(t *testing.T) {
+		c := New("", 0, nullLogger())
+
+		ctxWithValue := context.WithValue(context.Background(),
+			"X-Openai-Ratelimit-RequestPM-Embedding", []string{"50"})
+
+		rl := c.GetVectorizerRateLimit(ctxWithValue, fakeClassConfig{})
+		assert.Equal(t, 50, rl.LimitRequests)
+		assert.Equal(t, 50, rl.RemainingRequests)
+	})
+
+	t.Run("pass rate limit headers tokens", func(t *testing.T) {
+		c := New("", 0, nullLogger())
+
+		ctxWithValue := context.WithValue(context.Background(), "X-Openai-Ratelimit-TokenPM-Embedding", []string{"60"})
+
+		rl := c.GetVectorizerRateLimit(ctxWithValue, fakeClassConfig{})
+		assert.Equal(t, 60, rl.LimitTokens)
+		assert.Equal(t, 60, rl.RemainingTokens)
+	})
+}
+
+type fakeHandler struct {
+	t           *testing.T
+	serverError error
+}
+
+func (f *fakeHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	assert.Equal(f.t, http.MethodPost, r.Method)
+
+	if f.serverError != nil {
+		embeddingError := map[string]interface{}{
+			"message": f.serverError.Error(),
+			"type":    "invalid_request_error",
+		}
+		embedding := map[string]interface{}{
+			"error": embeddingError,
+		}
+		outBytes, err := json.Marshal(embedding)
+		require.Nil(f.t, err)
+
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Write(outBytes)
+		return
+	}
+
+	bodyBytes, err := io.ReadAll(r.Body)
+	require.Nil(f.t, err)
+	defer r.Body.Close()
+
+	var b map[string]interface{}
+	require.Nil(f.t, json.Unmarshal(bodyBytes, &b))
+
+	textInputArray := b["input"].([]interface{})
+	textInput := textInputArray[0].(string)
+	assert.Greater(f.t, len(textInput), 0)
+
+	embeddingData := map[string]interface{}{
+		"object":    textInput,
+		"index":     0,
+		"embedding": []float32{0.1, 0.2, 0.3},
+	}
+	embedding := map[string]interface{}{
+		"object": "list",
+		"data":   []interface{}{embeddingData},
+	}
+
+	outBytes, err := json.Marshal(embedding)
+	require.Nil(f.t, err)
+
+	w.Write(outBytes)
+}
+
+func nullLogger() logrus.FieldLogger {
+	l, _ := test.NewNullLogger()
+	return l
+}
+
+// TODO: Check if Databricks API has error codes
```

Comment:
can we remove this part? or uncomment if it's relevant?

Reply:
I'm actually waiting for a response form our partner about the error codes as well as other things in the TODO comments. I'll remove them once I hear back and make the necessary changes.